### DISCUSSION TITLE
how the Topic Embeddings in BERTopic are calculated?

### DISCUSSION THREAD

**Discussion by chonchonj23** [MaartenGr/BERTopic#1413] [Q&A]
Dear Maarten,

From Step 1 of BERTopic, we can get some document embeddings from a pre-trained Transformer model.

Then in the visualization step ( using, say, visualize_heatmap() ), how are those topic embeddings calculated (to further calculate the cosine similarity between topics)?

I feel that the n_clusters parameter in visualize_heatmap() can affect the cosine similarity scores, but I don't know if that also affects the calculation of topic embeddings.

Thank you very much for your help.

Best regards,
Chon

**Comment by MaartenGr** [comment]
The topic embeddings are the average of all embeddings within a topic. The `n_clusters` parameter in visualize_heatmap does not affect the calculation of the topic embeddings. It merely makes sure that similar topics will be shown together in the heatmap.

**Reply by chonchonj23** [reply]
Thank you very much Maarten.

BERTopic is awesome!
